log/log.go

```diff
@@ -1,159 +1,39 @@
 package log
 
 import (
-	"fmt"
-	"io"
 	"os"
-	"sync"
 
-	"github.com/ViaQ/logerr/kverrors"
+	"github.com/ViaQ/logerr/v2/internal/sink"
 	"github.com/go-logr/logr"
 )
 
-// ErrUnknownLoggerType is returned when trying to perform a *Logger only function
-// that is incompatible with logr.Logger interface
-var ErrUnknownLoggerType = kverrors.New("unknown error type")
-
-var (
-	defaultOutput  io.Writer = os.Stdout
-	defautLogLevel           = 0
-
-	// logLevel sets the level at which you want logs to be displayed
-	// By default the verbosity is set to 0 and all logs that do not
-	// use V(...) will be printed. To increase logging verbosity
-	logLevel = defautLogLevel
-
-	mtx    sync.RWMutex
-	logger logr.Logger = NewLogger("", os.Stdout, 0, JSONEncoder{})
-)
-
-// Init initializes the logger. This is required to use logging correctly
-// component is the name of the component being used to log messages.
-// Typically this is your application name.
-//
-// keyValuePairs are key/value pairs to be used with all logs in the future
-func Init(component string, keyValuePairs ...interface{}) {
-	InitWithOptions(component, nil, keyValuePairs...)
-}
-
-// MustInit calls Init and panics if it returns an error
-func MustInit(component string, keyValuePairs ...interface{}) {
-	Init(component, keyValuePairs...)
+// NewLogger creates a logger with the provided key value pairs
+func NewLogger(component string, keyValuePairs ...interface{}) logr.Logger {
+	return NewLoggerWithOptions(component, nil, keyValuePairs...)
 }
 
-// InitWithOptions inits the logger with the provided opts
-func InitWithOptions(component string, opts []Option, keyValuePairs ...interface{}) {
-	mtx.Lock()
-	defer mtx.Unlock()
-
-	ll := NewLogger(component, defaultOutput, 0, JSONEncoder{}, keyValuePairs...)
+// NewLoggerWithOptions creates a logger with the provided opts and key value pairs
+func NewLoggerWithOptions(component string, opts []Option, keyValuePairs ...interface{}) logr.Logger {
+	sink := sink.NewLogSink(component, os.Stdout, 0, sink.JSONEncoder{}, keyValuePairs...)
 
 	for _, opt := range opts {
-		opt(ll)
+		opt(sink)
 	}
 
-	// don't lock because we already have a lock
-	useLogger(ll)
-}
-
-// MustInitWithOptions calls InitWithOptions and panics if an error is returned
-func MustInitWithOptions(component string, opts []Option, keyValuePairs ...interface{}) {
-	InitWithOptions(component, opts, keyValuePairs...)
+	return logr.New(sink)
 }
 
-// GetLogger returns the root logger used for logging
-func GetLogger() logr.Logger {
-	return logger
-}
+// SetOptions sets the options of the logger if the `logr.Sink` is a `Sink`
+func SetOptions(l logr.Logger, opts []Option) bool {
```

Comment:
Does it make sense to have `opts` as a varargs parameter for additional ease of use?

Reply:
See comment below.